PWGEM/Dilepton/Core/Dilepton.h

```diff
@@ -805,15 +825,26 @@ struct Dilepton {
     ROOT::Math::PtEtaPhiMVector v2(t2.pt(), t2.eta(), t2.phi(), leptonM2);
     ROOT::Math::PtEtaPhiMVector v12 = v1 + v2;
 
+    float dca3D_t1 = 999.f, dca3D_t2 = 999.f, pair_dca3D = 999.f;
+    float dcaxy_t1 = 999.f, dcaxy_t2 = 999.f, pair_dcaxy = 999.f;
+    float dcaz_t1 = 999.f, dcaz_t2 = 999.f, pair_dcaz = 999.f;
     float dca_t1 = 999.f, dca_t2 = 999.f, pair_dca = 999.f;
     if constexpr (pairtype == o2::aod::pwgem::dilepton::utils::pairutil::DileptonPairType::kDielectron) {
-      dca_t1 = dca3DinSigma(t1);
-      dca_t2 = dca3DinSigma(t2);
-      pair_dca = std::sqrt((dca_t1 * dca_t1 + dca_t2 * dca_t2) / 2.);
-      if (cfgUseDCAxy) {
-        dca_t1 = t1.dcaXY() / std::sqrt(t1.cYY());
-        dca_t2 = t2.dcaXY() / std::sqrt(t2.cYY());
-        pair_dca = std::sqrt((dca_t1 * dca_t1 + dca_t2 * dca_t2) / 2.);
+      dca3D_t1 = dca3DinSigma(t1);
+      dca3D_t2 = dca3DinSigma(t2);
+      pair_dca3D = std::sqrt((dca3D_t1 * dca3D_t1 + dca3D_t2 * dca3D_t2) / 2.);
+      pair_dca = pair_dca3D;
+      if (cfgDCAcase == 1 || cfgDCAcase == 3) {
+        dcaxy_t1 = t1.dcaXY() / std::sqrt(t1.cYY());
+        dcaxy_t2 = t2.dcaXY() / std::sqrt(t2.cYY());
+        pair_dcaxy = std::sqrt((dcaxy_t1 * dcaxy_t1 + dcaxy_t2 * dcaxy_t2) / 2.);
+        pair_dca = pair_dcaxy;
```

Comment:
Using this syntax makes case `cfgDCAcase = 3` obsolete. You only fill the histogram once.